datafusion/core/src/datasource/file_format/mod.rs

```diff
@@ -106,6 +131,67 @@ pub trait FileFormat: Send + Sync + fmt::Debug {
     }
 }
 
+/// A container of [FileFormatFactory] which also implements [FileType].
+/// This enables converting a dyn FileFormat to a dyn FileType.
+/// The former trait is a superset of the latter trait, which includes execution time
+/// relevant methods. [FileType] is only used in logical planning and only implements
+/// the subset of methods required during logical planning.
+pub struct DefaultFileFormat {
+    file_format_factory: Arc<dyn FileFormatFactory>,
+}
+
+impl DefaultFileFormat {
+    /// Constructs a [DefaultFileFormat] wrapper from a [FileFormatFactory]
+    pub fn new(file_format_factory: Arc<dyn FileFormatFactory>) -> Self {
+        Self {
+            file_format_factory,
+        }
+    }
+}
+
+impl FileType for DefaultFileFormat {
```

Comment:
Rather than this wrapper, I wonder if we could implement `FileType` for `dyn FileFormatFactory`

```rust
impl FileType for &dyn FileFormatFactory{ 
...
}
```

And avoid having to have this struct 🤔

Reply:
I tried briefly, but ran into various errors. I am not sure if it is possible with the traits defined in different crates to make this work without a wrapper. I think that is why `DefaultTableSource` exists as well.